### PR TITLE
Make `For` component each prop work with signals

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Despite what the docs say, I have been unable to pass signals to the `For` component. To test this, try running the changed code for sso auth axum without the changes I made to the `For` component.

### CODE REVIEW

File: leptos/src/for_loop.rs [hunk 1]
```diff
@@ -24,19 +25,7 @@ use std::hash::Hash;
...
-///         each=move || counters.get()
-///         // a unique key for each item
-///         key=|counter| counter.id
-///         // renders each item to a view
-///         children=move |counter: Counter| {
-///           view! {
-///             <button>"Value: " {move || counter.count.get()}</button>
-///           }
-///         }
-///       />
```
Comment (SleeplessOne1917) [inline leptos/src/for_loop.rs:36]: This was repeated twice. I'm assuming that was unintentional. Let me know if you want to keep it.

**Comment by gbj** [comment]
I do not understand your statement of the problem. What is the code you are expecting to compile that doesn't? This function signature has been the same since ~0.1 so I'd be very surprised if it is broken somehow, rather than this being a misunderstanding.

**Comment by SleeplessOne1917** [comment]
> What is the code you are expecting to compile that doesn't?

Among other things, the example that I modified as part of this PR. If you try to compile that example by passing in the signal instead of a closure that calls 'get' on that signal using the code as it is in main, you will find that it doesn't compile.

**Comment by SleeplessOne1917** [comment]
I was not aware of the nightly limitation. Closing this PR.

**Comment by SleeplessOne1917** [comment]
Strangely, I ran into this issue in another project despite having a rust-toolchain.toml in the repo root that sets the channel to nightly.

**Comment by gbj** [comment]
Using the nightly features requires two things:
1) having the nightly compiler 
2) enabling the `nightly` feature on the Leptos crate

**Pull request closed without merging**